### PR TITLE
ENH: scipy.signal - Tukey window

### PR DESCRIPTION
**Pull request by e-q**
The Tukey window is helpful when you need to squeeze out as much as you can from limited data. I used [the definition on Wikipedia](http://en.wikipedia.org/wiki/Window_function#Tukey_window), and adapted the docstring from existing functions. 

I appreciate any and all comments and feedback!

### CODE REVIEW

File: scipy/signal/windows.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 __all__ = ['boxcar', 'triang', 'parzen', 'bohman', 'blackman', 'nuttall',
            'blackmanharris', 'flattop', 'bartlett', 'hanning', 'barthann',
            'hamming', 'kaiser', 'gaussian', 'general_gaussian', 'chebwin',
-           'slepian', 'cosine', 'hann', 'exponential', 'get_window']
+           'slepian', 'cosine', 'hann', 'exponential', 'tukey',  'get_window']
```
Comment (rgommers) [inline scipy/signal/windows.py]: need only one space after the comma

**Comment by rgommers** [comment]
It would be good to mention this new window on the mailing list. If there are more people that have a use for it, we can make sure that the API is OK for them

**Comment by rgommers** [comment]
A few more things to do:
- add a unit test
- add to function listing in ``signal/__init__.py``
- add to release notes.

See gh-4536 for a recent PR that added the ``exponential`` window.

File: scipy/signal/windows.py [hunk 2]
```diff
@@ -709,6 +709,91 @@ def hann(M, sym=True):
...
+    Returns
+    -------
+    w : ndarray
+        The window, with the maximum value normalized to 1 (though the value 1
+        does not appear if `M` is even and `sym` is True).
+
+    References
+    ----------
+    .. [1] Wikipedia, "Window function",
+           http://en.wikipedia.org/wiki/Window_function
```
Comment (rgommers) [inline scipy/signal/windows.py]: I'd add ``#Tukey_window`` at the end of this link. And maybe also a reference to the original article (Tukey, "An introduction to the calculations of numerical spectrum analysis")?

File: scipy/signal/windows.py [hunk 3]
```diff
@@ -709,6 +709,91 @@ def hann(M, sym=True):
...
 
+def tukey(M, alpha=0.5, sym=True):
+    r"""Return a Tukey window, also known as a tapered cosine window. 
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an empty
+        array is returned.
+    alpha : float
```
Comment (rgommers) [inline scipy/signal/windows.py]: add ``, optional``

**Comment by rgommers** [comment]
``pep8 scipy`` also gives a few errors (hence the TravisCI failure). You may want to run that command locally before sending/updating a PR.

**Comment by e-q** [comment]
Thanks for the feedback! I've made the necessary changes, and will let the mailing list know.

File: scipy/signal/windows.py [hunk 4]
```diff
@@ -709,6 +709,92 @@ def hann(M, sym=True):
...
+
+    Returns
+    -------
+    w : ndarray
+        The window, with the maximum value normalized to 1 (though the value 1
+        does not appear if `M` is even and `sym` is True).
+
+    References
+    ----------
+    .. [1] Tukey, J.W., (1967), "An introduction to the calculations of
```
Comment (ewmoore) [inline scipy/signal/windows.py]: Do you know the rest of this reference? Is it a book?

File: scipy/signal/windows.py [hunk 5]
```diff
@@ -709,6 +709,92 @@ def hann(M, sym=True):
...
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an empty
+        array is returned.
+    alpha : float, optional
+        Shape parameter of the Tukey window, representing the faction of the
+        window inside the cosine tapered region. 
+        If zero, the Tukey window is equivalent to a rectangular window.
+        If one, the Tukey window is equivalent to a Hann window.
```
Comment (ewmoore) [inline scipy/signal/windows.py]: How about adding a test that checks these statements.

File: scipy/signal/windows.py (hunk 4 above)
Comment (e-q) [inline scipy/signal/windows.py]: Not personally, I took it on faith from Wikipedia... Alternatively, I was able to easily access the other reference listed there, which contains a plethora of information on various windows, including the Tukey. I'll use that one instead.

File: scipy/signal/windows.py (hunk 5 above)
Comment (e-q) [inline scipy/signal/windows.py]: Good call, I've done so.

**Comment by ewmoore** [comment]
Sorry for not catching this earlier, but there is tiny change still needed in get_window.  On line 1636 of the file in your branch, there is a list of windows that require a parameter.  Tukey should be added to that list.

Personally I'd rather not add the 'tuk' abbreviation either, but looking at get_window that ship has clearly sailed.  So I guess I'm -0 on that.

**Comment by ewmoore** [comment]
If you'd like, you could also add the recently merged exponential window that same list, seems I missed that when reviewing the earlier PR.

**Comment by e-q** [comment]
Ah, ok. I did not include it there since there are default values for the parameters, and I assumed it would be fine to simply return a window with those defaults.

**Comment by e-q** [comment]
Also, I feel it may make sense for `get_window` to use `winstr.lower()` for the comparisons, in case someone inputs 'Hann' or something. Then again, if this issue has never arisen before, it may just be clutter...

**Comment by ewmoore** [comment]
In general, we should be less forgiving here.  There are already too many different strings that result in identical windows.  Let not add a call to `lower` and multiply that up even more.

**Comment by ewmoore** [comment]
@e-q, if you could squash this down to 1 or two commits, I'll merge it in a few days barring any further comments.

**Comment by e-q** [comment]
Squashed. Thanks!

**Comment by ewmoore** [comment]
Travis failed due to exceeding the allowed time.  I restarted that particular job.

**Comment by ewmoore** [comment]
Thanks @e-q!

**Pull request merged**